### PR TITLE
feat(inputs.filecount): Add oldestFileTimestamp and newestFileTimestamp

### PR DESCRIPTION
**Pull request by tguenneguez**
Add fields like :
oldest_file_timestamp
newest_file_timestamp

Use case:

With this two fields, we can generate alarming if nothing append in the directeory.
For exemple if we wait for daily file reception.

resolves https://github.com/influxdata/telegraf/issues/11087

### CODE REVIEW

**Comment by powersj** [comment]
@tguenneguez you can push additional commits to a PR and resolve tests :) no need to close and re-create PRs

**Comment by tguenneguez** [comment]
> @tguenneguez you can push additional commits to a PR and resolve tests :) no need to close and re-create PRs

Thanks a lot ;-)

**Comment by tguenneguez** [comment]
> @tguenneguez you can push additional commits to a PR and resolve tests :) no need to close and re-create PRs

Hello
I don't arrive to compile it on my own server. I this it's because of the proxy.
Can you help me to understand error and solve it ?
https://app.circleci.com/pipelines/github/influxdata/telegraf/16081/workflows/6cc459ee-ff59-491b-94c5-d936ae179b96/jobs/251757?invite=true#step-115-1106
And 
https://app.circleci.com/pipelines/github/influxdata/telegraf/16081/workflows/6cc459ee-ff59-491b-94c5-d936ae179b96/jobs/251756?invite=true#step-105-485

Thanks
Thomas

**Comment by tguenneguez** [comment]
When I test on my own server I have the error :
cloud.google.com/go/@v/v0.107.0.mod 404

**Comment by powersj** [comment]
> cloud.google.com/go/@v/v0.107.0.mod 404

Hmm that does not look like a valid package URL, hence the 404. Did you make any changes to the go.mod or go.sum?

It looks like it was probably trying to get to download the dependency: `cloud.google.com/go v0.107.0 // indirect`. If you were running this on the CLI the similar command would be `go get cloud.google.com/go@v0.107.0`

> https://app.circleci.com/pipelines/github/influxdata/telegraf/16081/workflows/6cc459ee-ff59-491b-94c5-d936ae179b96/jobs/251756?invite=true#step-105-485

This is an integration test error. It is most likely unrelated to your changes and can be ignored.

> https://app.circleci.com/pipelines/github/influxdata/telegraf/16081/workflows/6cc459ee-ff59-491b-94c5-d936ae179b96/jobs/251757?invite=true#step-115-1106

These are two unit test failures on the `inputs.filecount` plugin. Specifically, `TestNoRecursiveButSuperMeta` and `TestNoFiltersOnChildDir`. You modified these tests to check for the file timestamp, but you did not set `FileTimestamp` to True. Meaning that those fields are not present.

**Comment by tguenneguez** [comment]
> > cloud.google.com/go/@v/v0.107.0.mod 404
> 
> Hmm that does not look like a valid package URL, hence the 404. Did you make any changes to the go.mod or go.sum?
> 
> It looks like it was probably trying to get to download the dependency: `cloud.google.com/go v0.107.0 // indirect`. If you were running this on the CLI the similar command would be `go get cloud.google.com/go@v0.107.0`
> 
> > https://app.circleci.com/pipelines/github/influxdata/telegraf/16081/workflows/6cc459ee-ff59-491b-94c5-d936ae179b96/jobs/251756?invite=true#step-105-485
> 
> This is an integration test error. It is most likely unrelated to your changes and can be ignored.
> 
> > https://app.circleci.com/pipelines/github/influxdata/telegraf/16081/workflows/6cc459ee-ff59-491b-94c5-d936ae179b96/jobs/251757?invite=true#step-115-1106
> 
> These are two unit test failures on the `inputs.filecount` plugin. Specifically, `TestNoRecursiveButSuperMeta` and `TestNoFiltersOnChildDir`. You modified these tests to check for the file timestamp, but you did not set `FileTimestamp` to True. Meaning that those fields are not present.

Thanks a lot

**Comment by tguenneguez** [comment]
@powersj : It's done ;-)

**Review by powersj — APPROVED** [review]
Thanks for the PR. My only open question is if we should put this behind an option. Adding new fields usually isn't that big of a deal so I'll give it a +1 for Sven to give a second opinion.

**Comment by tguenneguez** [comment]
> Thanks for the PR. My only open question is if we should put this behind an option. Adding new fields usually isn't that big of a deal so I'll give it a +1 for Sven to give a second opinion.

Originally I was thinking of putting an option, but since the information is easy to present, if someone wants to disable them as much as they use:
fielddrop

**Review by srebhan — APPROVED** [review]
LGTM. Thanks for your contribution @tguenneguez! Regarding adding new fields we have not been consistent in the past so I don't want to hold up this OR.

**Pull request merged**
